File: ERPSystem.Application/Common/GeneralManagerAccess.cs
namespace ERPSystem.Application.Common;

/// <summary>
/// General manager (مدير عام) — exclusive access to import/cost pricing and the China module.
/// </summary>
public static class GeneralManagerAccess
{
    public const string PermissionCode = "security.general-manager";

    public static bool IsGeneralManager(IReadOnlyCollection<string> permissionCodes) =>
        permissionCodes.Any(code => code.Equals(PermissionCode, StringComparison.OrdinalIgnoreCase));
}

File: web-client/src/auth/generalManagerAccess.ts
export const GENERAL_MANAGER_PERMISSION = 'security.general-manager';

export function isGeneralManager(permissions: readonly string[]): boolean {
  return permissions.some((code) => code.toLowerCase() === GENERAL_MANAGER_PERMISSION);
}

export function canViewSensitivePricing(permissions: readonly string[]): boolean {
  return isGeneralManager(permissions);
}
